src/pybind/mgr/volumes/fs/stats_util.py

```diff
@@ -54,3 +75,246 @@ def get_stats(src_path, dst_path, fs_handle):
         'amount cloned': get_size_ratio_str(size_c, size_t),
         'files cloned': get_num_ratio_str(rentries_c, rentries_t),
     }
+
+
+class CloneInfo:
+
+    def __init__(self, volname):
+        self.volname = volname
+
+        self.src_group_name = None
+        self.src_subvol_name = None
+        self.src_path = None
+
+        self.dst_group_name = None
+        self.dst_subvol_name = None
+        self.dst_path = None
+
+
+class CloneProgressReporter:
```

Comment:
>Why are most routines in this class starting with "_". Is it to denote them as internal?

Yep, in Python prefixing methods names with underscore are used to tell users which methods are internal and which are not. `initiate_reporting()` doesn't have underscore because it's meant for code outside this class.

Reply:
> Also, most of the log message are at info level. Please audit those to check if some of them can be moved to debug log level.

Sure, I'll get to it.

Reply:
> > Why are most routines in this class starting with "_". Is it to denote them as internal?
> 
> Yep, in Python prefixing methods names with underscore are used to tell users which methods are internal and which are not. `initiate_reporting()` doesn't have underscore because it's meant for code outside this class.

Is that the convention followed in other parts of the volumes module? If not, we should stick to what's followed.

Reply:
Okay, looked for and found few instances -
 
 ```
def _create_subvolume(self, fs_handle, volname, group, subvolname, **kwargs)
def create_subvolume(self, **kwargs):
```

```
def _clone_subvolume_snapshot(self, fs_handle, volname, s_group, s_subvolume, **kwargs): 
def clone_subvolume_snapshot(self, **kwargs):
 ```
 
 ```
  def _track(self, sink_path):  
  def track(self, sink_path):
  ```

Reply:
Also, this is a Python-wide convention...